internal/controller/snapshot/snapshot_adapter.go

```diff
@@ -1058,3 +1068,122 @@ func (a *Adapter) findSnapshotWithOpenedPR(snapshot *applicationapiv1alpha1.Snap
 	}
 	return nil, nil
 }
+
+// checkOldSnapshotsPipelineRun sorts all snapshots for application and cancells all running interationTest pipelineruns within application
+// removes finalizer before the pipelinerun is set as cancelled run finally to be gracefuly cancelled
+// god please forgive me for this monstrosity of cognitive complexity 40+
+func (a *Adapter) checkAndCancelOldSnapshotsPipelineRun(application *applicationapiv1alpha1.Application, snapshot *applicationapiv1alpha1.Snapshot) error {
```

Comment:
We should also first check if this is a PR Snapshot before doing anything in this function (or maybe before even calling it) because otherwise we'd just be wasting resources by fetching Snapshots and looping through them.

Reply:
Agreed, set a condition before the function gets executed.